WebContent/tetsgame.js

```diff
@@ -1,123 +1,188 @@
+
+function createBlock(){
+	var num = Math.floor(Math.random()*8);
+	var block;
+	if  (num == 0) { activeBlock = new Block([[4,0],[5,0],[5,1]],"pink")
+		} else if (num == 1) { activeBlock = new Block([[5,0],[5,1],[5,2],[5,3]],"cyan")
+		} else if (num == 2) { activeBlock = new Block([[4,0],[5,0],[5,1],[5,2]],"blue")
+		} else if (num == 3) { activeBlock = new Block([[5,0],[5,1],[5,2],[4,2]],"orange")
+		} else if (num == 4) { activeBlock = new Block([[5,0],[4,1],[5,1],[4,2]],"red")
+		} else if (num == 5) { activeBlock = new Block([[4,1],[5,0],[5,1],[5,2]],"purple")
+		} else if (num == 6) { activeBlock = new Block([[4,0],[5,0],[5,1],[6,1]],"green")
+		} else if (num == 7) { activeBlock = new Block([[4,0],[4,1],[5,0],[5,1]],"yellow")
+	} return activeBlock; 
+};
+
 /* time step */
- gameboard();
 	
-var time =setInterval(function (){ blockmove(block)}, 5000);
-function blockmove () {
-	for (i=0; i<block.length; i++) {
-	block[i][1] += 1;
+var time =setInterval(function (){ blockmove(activeBlock)}, 2000);
+function blockmove() {
+	var shiftedCoordinates = activeBlock.coordinates.map(function(coordinate){
+		return [
+		    coordinate[0],
+		    coordinate[1]+1
+		];
+	});
+	if (grid.willFit(shiftedCoordinates)){
+		for (var i=0; i<activeBlock.coordinates.length; i++) {
+			activeBlock.coordinates[i][1] += 1;
+		}
+	} else {activeBlock.active = false;
+	console.log(activeBlock.coordinates)
 	}
 	gameboard();
-return block 
+return activeBlock 
 }
 
-var testblock =[[0,0],[1,0]];
-var block1 = [[0,0],[0,1,],[0,2],[0,3]]; 
-var block2 =[[0,0],[1,0],[1,1],[1,2]];
-var block3 = [[1,0],[1,1],[1,2],[0,2]];
-var block4 = [[1,0],[0,1],[1,1],[0,2]];
-var block5 = [[0,1],[1,0],[1,1],[1,2]];
-var block6 = [[0,0],[1,0],[1,1],[1,2]];
-var block7 =[[0,0],[0,1],[1,0],[1,1]]; 
-
-/* block coordinates equal creategrid coordations make 0 be 1 */
-var num = Math.floor(Math.random()*8);
-if  (num ==0) { 
-	var block = testblock; 
-	var z =1;
-} else if (num == 1) {
-	var block = block1;
-	var z = 2;
-} else if (num == 2) {
-	var block = block2;
-	var z = 3;
-} else if (num == 3) {
-	var block = block3;
-	var z = 4;
-} else if (num == 4) {
-	var block = block4;
-	var z = 5;
-} else if (num == 5) {
-	var block = block5;
-	var z = 6;
-} else if (num == 6) {
-	var block = block6;
-	var z = 7;
-} else if (num == 7) {
-	var block = block7;
-	var z = 8;
-	
-};
 
-
-function addblock(block,z) {
-	for (i=0; i<block.length; i++){
-	 x=block[i][0];
-	 y=block[i][1];
-	 grid[y][x] = z ;
-	 }
+function addblock(activeBlock) {
+	for (var i=0; i<activeBlock.coordinates.length; i++){
+		x=activeBlock.coordinates[i][0];
+		y=activeBlock.coordinates[i][1];
+		context.beginPath();					// draws block onto grid
+		context.fillStyle = activeBlock.colour; 
+		context.rect(x*20, (y-2)*20, 20, 20);  
+		context.fill(); 
+		}
 }
 
 
 
+
 /* keyboard functions */
-function moveleft(block) {
-	for (i=0; i<block.length; i++) {
-	block[i][0] -= 1;
+function moveleft(activeBlock) {
```

Comment:
If you look at the code inside ```blockMove```, ```moveleft``` and ```moveright```, it is all the same except for what is added to each of the coordinates.

Perhaps there could be a general function ```moveBlock(block, direction)```, where direction is a unit vector that we want to move the block by (i.e. to move right: ```moveBlock(activeBlock, [1,0]);```)